From 446d33d5595f6cc2c65ca95a9008146df762ad94 Mon Sep 17 00:00:00 2001
From: Yoshisato Yanagisawa <yyanagisawa@chromium.org>
Date: Thu, 14 Dec 2023 23:28:49 +0000
Subject: [PATCH] Ensure `context_` exists.

This is a bug fix for crbug.com/1157715.
The `context_` was accessed even if it becomes invalid, which caused the
crash in `content::EmbeddedWorkerInstance::OnWorkerVersionDoomed()`.

Bug: 1157715
Change-Id: Ib02aa17683a0d5dfd549ca1c2ffd515a54f1200b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5117771
Commit-Queue: Yoshisato Yanagisawa <yyanagisawa@chromium.org>
Reviewed-by: Shunya Shishido <sisidovski@chromium.org>
Reviewed-by: Minoru Chikamune <chikamune@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1237809}
---

--- a/content/browser/service_worker/embedded_worker_instance.cc
+++ b/content/browser/service_worker/embedded_worker_instance.cc
@@ -659,6 +659,9 @@
 }
 
 void EmbeddedWorkerInstance::OnWorkerVersionDoomed() {
+  if (!context_) {
+    return;
+  }
   ServiceWorkerDevToolsManager::GetInstance()->WorkerVersionDoomed(
       process_id(), worker_devtools_agent_route_id(),
       base::WrapRefCounted(context_->wrapper()), owner_version_->version_id());
@@ -1115,6 +1118,7 @@
 mojo::PendingRemote<network::mojom::URLLoaderFactory>
 EmbeddedWorkerInstance::MakeScriptLoaderFactoryRemote(
     std::unique_ptr<blink::PendingURLLoaderFactoryBundle> script_bundle) {
+  CHECK(context_);
   mojo::PendingRemote<network::mojom::URLLoaderFactory>
       script_loader_factory_remote;
 
